### PR TITLE
fix: Fixes white space comparision issues on run test.

### LINKED ISSUE #34344: Tests for various challenges are too strict about whitespace around function arguments
## The problem
In the "[React and Redux: Moving Forward From Here](https://learn.freecodecamp.org/front-end-libraries/react-and-redux/moving-forward-from-here)" challenge, I tried the following code:
```js
console.log( 'Now I know React and Redux!' );
```

However, this code did not pass the tests. But if I simply remove the spaces, it passes:
```js
console.log('Now I know React and Redux!');
```

I think the tests for this challenge need to be less strict on spacing. Perhaps a partial minifier could be run on the code to remove excess whitespace within lines before making the test comparison?

**This same kind of issue also affects the following challenges:**
- ["Data Visualization with D3: Add Axes to a Visualization"](https://learn.freecodecamp.org/data-visualization/data-visualization-with-d3/add-axes-to-a-visualization): spaces are not allowed around `call` method.
```js
svg
      .append( 'g' )
      .attr(
        'transform',
        `translate(${padding}, 0)`
      )
      .call( yAxis );
```
- ["JSON APIs and Ajax: Get JSON with the JavaScript XMLHttpRequest Method"](https://learn.freecodecamp.org/data-visualization/json-apis-and-ajax/get-json-with-the-javascript-xmlhttprequest-method): spaces not allowed around `JSON.parse` arguments.
```js
const apiRequest = new XMLHttpRequest();
  apiRequest.open( 'GET', '/json/cats.json', true );
  apiRequest.send();
  apiRequest.onload = function() {
    const json = JSON.parse( apiRequest.responseText );
    document.getElementsByClassName( 'message' )[0].innerHTML = JSON.stringify( json );
  };
```

## Browser and OS info
* Browser Name: Firefox Developer Edition
* Browser Version: 66.0b5
* Operating System: Antergos Linux

### PR DESCRIPTION
**Pull request by wmollenkopf**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #34344

### CODE REVIEW

File: curriculum/challenges/arabic/03-front-end-libraries/react-and-redux/moving-forward-from-here.arabic.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ undefined
 ```yml
 tests:
   - text: ''
-    testString: 'assert(editor.getValue().includes("console.log("Now I know React and Redux!")") || editor.getValue().includes("console.log(\"Now I know React and Redux!\")"), "The message <code>Now I know React and Redux!</code> should be logged to the console.");'
+    testString: assert(new RegExp(/console.log\((\s)*\"Now I know React and Redux!\"(\s)*\)\;/g).test(editor.getValue()) || new RegExp(/console.log\((\s)*\'Now I know React and Redux!\'(\s)*\)\;/g).test(editor.getValue()), 'The message <code>Now I know React and Redux!</code> should be logged to the console.');
```
Comment (RandellDawson) [inline curriculum/challenges/arabic/03-front-end-libraries/react-and-redux/moving-forward-from-here.arabic.md]:
```suggestion
    testString: assert(/console\.log\(\s*("|')Now I know React and Redux!\1\s*\);?/.test(code));
```

File: curriculum/challenges/chinese/03-front-end-libraries/react-and-redux/moving-forward-from-here.chinese.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ localeTitle: 从这里前进
 ```yml
 tests:
   - text: 消息<code>Now I know React and Redux!</code>应该登录到控制台。
-    testString: 'assert(editor.getValue().includes("console.log("Now I know React and Redux!")") || editor.getValue().includes("console.log(\"Now I know React and Redux!\")"), "The message <code>Now I know React and Redux!</code> should be logged to the console.");'
+    testString: assert(new RegExp(/console.log\((\s)*\"Now I know React and Redux!\"(\s)*\)\;/g).test(editor.getValue()) || new RegExp(/console.log\((\s)*\'Now I know React and Redux!\'(\s)*\)\;/g).test(editor.getValue()), 'The message <code>Now I know React and Redux!</code> should be logged to the console.');
```
Comment (RandellDawson) [inline curriculum/challenges/chinese/03-front-end-libraries/react-and-redux/moving-forward-from-here.chinese.md]:
```suggestion
    testString: assert(/console\.log\(\s*("|')Now I know React and Redux!\1\s*\);?/.test(code));
```

File: curriculum/challenges/english/03-front-end-libraries/react-and-redux/moving-forward-from-here.english.md [hunk 3]
```diff
@@ -24,7 +24,7 @@ Log the message <code>'Now I know React and Redux!'</code> to the console.
 ```yml
 tests:
   - text: The message <code>Now I know React and Redux!</code> should be logged to the console.
-    testString: assert(editor.getValue().includes('console.log("Now I know React and Redux!")') || editor.getValue().includes('console.log(\'Now I know React and Redux!\')'), 'The message <code>Now I know React and Redux!</code> should be logged to the console.');
+    testString: assert(new RegExp(/console.log\((\s)*\"Now I know React and Redux!\"(\s)*\)\;/g).test(editor.getValue()) || new RegExp(/console.log\((\s)*\'Now I know React and Redux!\'(\s)*\)\;/g).test(editor.getValue()), 'The message <code>Now I know React and Redux!</code> should be logged to the console.');
```
Comment (RandellDawson) [inline curriculum/challenges/english/03-front-end-libraries/react-and-redux/moving-forward-from-here.english.md]:
```suggestion
    testString: assert(/console\.log\(\s*("|')Now I know React and Redux!\1\s*\);?/.test(code));
```

File: curriculum/challenges/portuguese/03-front-end-libraries/react-and-redux/moving-forward-from-here.portuguese.md [hunk 4]
```diff
@@ -19,7 +19,7 @@ localeTitle: Avançando daqui
 ```yml
 tests:
   - text: A mensagem <code>Now I know React and Redux!</code> deve ser registrado no console.
-    testString: 'assert(editor.getValue().includes("console.log("Now I know React and Redux!")") || editor.getValue().includes("console.log(\"Now I know React and Redux!\")"), "The message <code>Now I know React and Redux!</code> should be logged to the console.");'
+    testString: assert(new RegExp(/console.log\((\s)*\"Now I know React and Redux!\"(\s)*\)\;/g).test(editor.getValue()) || new RegExp(/console.log\((\s)*\'Now I know React and Redux!\'(\s)*\)\;/g).test(editor.getValue()), 'The message <code>Now I know React and Redux!</code> should be logged to the console.');
```
Comment (RandellDawson) [inline curriculum/challenges/portuguese/03-front-end-libraries/react-and-redux/moving-forward-from-here.portuguese.md]:
```suggestion
    testString: assert(/console\.log\(\s*("|')Now I know React and Redux!\1\s*\);?/.test(code));
```

File: curriculum/challenges/russian/03-front-end-libraries/react-and-redux/moving-forward-from-here.russian.md [hunk 5]
```diff
@@ -19,7 +19,7 @@ localeTitle: Перемещение вперед
 ```yml
 tests:
   - text: Сообщение <code>Now I know React and Redux!</code> должен быть зарегистрирован на консоли.
-    testString: 'assert(editor.getValue().includes("console.log("Now I know React and Redux!")") || editor.getValue().includes("console.log(\"Now I know React and Redux!\")"), "The message <code>Now I know React and Redux!</code> should be logged to the console.");'
+    testString: assert(new RegExp(/console.log\((\s)*\"Now I know React and Redux!\"(\s)*\)\;/g).test(editor.getValue()) || new RegExp(/console.log\((\s)*\'Now I know React and Redux!\'(\s)*\)\;/g).test(editor.getValue()), 'The message <code>Now I know React and Redux!</code> should be logged to the console.');
```
Comment (RandellDawson) [inline curriculum/challenges/russian/03-front-end-libraries/react-and-redux/moving-forward-from-here.russian.md]:
```suggestion
    testString: assert(/console\.log\(\s*("|')Now I know React and Redux!\1\s*\);?/.test(code));
```

File: curriculum/challenges/spanish/03-front-end-libraries/react-and-redux/moving-forward-from-here.spanish.md [hunk 6]
```diff
@@ -19,7 +19,7 @@ localeTitle: Avanzando desde aquí
 ```yml
 tests:
   - text: El mensaje ¡ <code>Now I know React and Redux!</code> debe estar registrado en la consola.
-    testString: 'assert(editor.getValue().includes("console.log("Now I know React and Redux!")") || editor.getValue().includes("console.log(\"Now I know React and Redux!\")"), "The message <code>Now I know React and Redux!</code> should be logged to the console.");'
+    testString: assert(new RegExp(/console.log\((\s)*\"Now I know React and Redux!\"(\s)*\)\;/g).test(editor.getValue()) || new RegExp(/console.log\((\s)*\'Now I know React and Redux!\'(\s)*\)\;/g).test(editor.getValue()), 'The message <code>Now I know React and Redux!</code> should be logged to the console.');
```
Comment (RandellDawson) [inline curriculum/challenges/spanish/03-front-end-libraries/react-and-redux/moving-forward-from-here.spanish.md]:
```suggestion
    testString: assert(/console\.log\(\s*("|')Now I know React and Redux!\1\s*\);?/.test(code));
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]

**Comment by RandellDawson** [comment]
@wmollenkopf I pulled your PR locally and once I rebased it, had to resolve a merge conflict.  After doing so, your changes do not work.  They fail for even the most basic solutions of:
```
console.log('Now I know React and Redux!');
```
See my suggested assert which should work.  You will have to update this same assert across all the language files.

**Comment by lynxlynxlynx** [comment]
If you plan to fix this PR, please reopen it with the working changes — I'm closing it due to lack of activity.

**Pull request closed without merging**